File: solutions/go/wordy/wordy.go
package wordy

import (
	"regexp"
	"strconv"
)

var pattern = regexp.MustCompile(`What is (-?\d+)( (plus|minus|multiplied by|divided by) -?\d+)*\?`)
var tokens = regexp.MustCompile(`-?\d+|is|plus|minus|multiplied by|divided by`)

func Answer(question string) (int, bool) {
	if !pattern.MatchString(question) {
		return 0, false
	}
	var result int
	var operator string

	for i, token := range tokens.FindAllString(question, -1) {
		if i%2 == 0 {
			operator = token
			continue
		}
		number, _ := strconv.Atoi(token)
		switch operator {
		case "is":
			result = number
		case "plus":
			result += number
		case "minus":
			result -= number
		case "multiplied by":
			result *= number
		case "divided by":
			result /= number
		}
	}
	return int(result), true
}

File: solutions/python/wordy/wordy.py
import re

_OPERATIONS = {
    "What is": lambda x, y: y,
    "plus": lambda x, y: x + y,
    "minus": lambda x, y: x - y,
    "multiplied by": lambda x, y: x * y,
    "divided by": lambda x, y: x // y,
}
_OPERATION_PATTERN = r"(?P<operation>\w+(?: (?:by|is))?)(?P<number> -?\d+)?"


def answer(question: str) -> int:
    result = 0
    for match in re.finditer(_OPERATION_PATTERN, question):
        operation = match.group('operation')
        number = match.group('number')

        if operation == "cubed":
            raise ValueError("unknown operation")
        if not operation or not number:
            raise ValueError("syntax error")
        # if operation not in _OPERATIONS:
        #     raise ValueError("unknown operation")

        result = _OPERATIONS[operation](result, int(number))

    return result
